components/brave_vpn/brave_vpn_connection_manager_mac.mm

```diff
@@ -0,0 +1,249 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_vpn/brave_vpn_connection_manager_mac.h"
+
+#import <Foundation/Foundation.h>
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/mac/bundle_locations.h"
+#include "base/mac/foundation_util.h"
+#include "base/strings/sys_string_conversions.h"
+
+// Referenced GuardianConnect implementation.
+// https://github.com/GuardianFirewall/GuardianConnect
+namespace brave_vpn {
```

Comment:
You should add (for all `.mm` files): 
```C++
#if !defined(__has_feature) || !__has_feature(objc_arc)
#error "This file requires ARC support."
#endif
```

right above the namespace declaration. It will make sure that the file itself is actually being compiled with ARC. Otherwise you might have memory leaks.

Reply:
Hmm,  `__has_feature(objc_arc)` gives `false`.
I got `error: "This file requires ARC support."`.

Reply:
Sorry for the late reply here, but that won't fix it as any Objective-C code would leak memory. IE: Strings, classes, etc.
You should fix it by doing: `configs += [ "//build/config/compiler:enable_arc" ]` in `BUILD.gn` file. This will enable "ARC" (Automatic Reference Counting) so you never have to worry about `release`, `dealloc`, `retain`, etc..

Reply:
I'm hesitating to enable `ARC` for macOS target because upstream doesn't use it now.
It looks like only iOS target uses `ARC` in chromium.

Reply:
Applied arc to brave_vpn target. https://github.com/brave/brave-core/pull/9246/commits/ff105233304fd4ac46a5494ee74f06ee7b9eb111